packages/next/src/build/index.ts

```diff
@@ -2258,24 +2287,25 @@ export default async function build(
               appConfig.revalidate === 0 ||
               exportResult.byPath.get(page)?.revalidate === 0
 
-            if (hasDynamicData && pageInfos.get(page)?.isStatic) {
+            const { isStatic, isPPR } = pageInfoRegistry.get(page, true)
+            if (hasDynamicData && isStatic) {
               // if the page was marked as being static, but it contains dynamic data
               // (ie, in the case of a static generation bailout), then it should be marked dynamic
-              pageInfos.set(page, {
-                ...(pageInfos.get(page) as PageInfo),
+              pageInfoRegistry.patch(page, {
                 isStatic: false,
                 isSSG: false,
               })
             }
 
             const isRouteHandler = isAppRouteRoute(originalAppPath)
 
-            // When this is an app page and PPR is enabled, the route supports
-            // partial pre-rendering.
-            const experimentalPPR =
-              !isRouteHandler && config.experimental.ppr === true
-                ? true
-                : undefined
+            /**
+             * If this page is not an app route and PPR is enabled, then it
+             * could support PPR.
+             */
+            const supportsPPR = !isRouteHandler && nextConfigPPREnabled
+
+            const experimentalPPR = isPPR ? true : undefined
```

Comment:
We're now using the information yielded from the static analysis to determine if the page is static or not rather than duplicating the logic here.